Vostok.Hosting/Components/BuildContext.cs

```diff
@@ -54,6 +54,7 @@ public BuildContext()
         public IVostokConfigurationSetupContext ConfigurationSetupContext { get; set; }
         public IVostokHostExtensions HostExtensions { get; set; }
         public List<object> DisposableHostExtensions { get; set; }
+        public EventLevelCounter EventLevelCounter { get; set; }

```

Comment:
Results of our verbal discussion:
- Get rid of diagnostic info provider (it doesn't bring any value on top of app metrics);
- Get rid of separate configuration builder;
- Get rid of internal caching;
- Expose a counter factory (we'll create one for vostok metrics and one for Houston instrumentation);
- Use `MultiFuncGauge` instead of implementing `IScrapableMetric`;
- Use `Instance` metric scope instread of `Application`;
- `BuildContext`: put counter closer to logs (use `Logs` class?);